### PR TITLE
Project TODOS- Elin Leidstedt 

### PR DESCRIPTION
**Pull request by Leidstedt**
Link to my deployed project: https://app.netlify.com/sites/affectionate-borg-3fcc3c/overview

My reflections on how this week's project turned out: Better than expected... 

Things I'd like to have clarified or explained in more detail: The "flow" of data... is it possible to visualize it somehow?

### CODE REVIEW

File: code/src/App.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import React from 'react'
+import { Home } from './pages/Home'
 
 export const App = () => {
   return (
     <div>
-      Find me in src/app.js!
+      <Home />
```
Comment (JennieDalgren) [inline code/src/App.js:7]: you don't need to wrap the `<Home />` component in divs since it's just one.

File: code/src/components/AddTaskForm.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+
+
+export const AddTaskForm = () => {
+    const dispatch = useDispatch()
+
+    const [inputValue, setInputValue] = useState("")
+
+    const handleOnSubmit = (e) => {
+        console.log(inputValue)
```
Comment (JennieDalgren) [inline code/src/components/AddTaskForm.js]: remove console log. 
Also, maybe add a check to see if the input is empty or not before adding it. Right now you can save an empty todo

File: code/src/components/AddTaskForm.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        )
+        setInputValue("")
+    }
+
+return (
+    <section className=''>
+    <h2>Add task</h2> 
+    <div>  
+    <form onSubmit={handleOnSubmit}>         
+        <lable>
```
Comment (JennieDalgren) [inline code/src/components/AddTaskForm.js]: <label> not <lable>

File: code/src/components/AddTaskForm.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        dispatch(todolist.actions.addTodo({
+            description: inputValue,
+            done: false
+        })
+        )
+        setInputValue("")
+    }
+
+return (
+    <section className=''>
```
Comment (JennieDalgren) [inline code/src/components/AddTaskForm.js]: remove className of not used

File: code/src/components/Header.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import { headerStyle } from 'components/headerStyle.css'
+import { ListSummary } from './ListSummary'
+
+export const Header = () => {
+  
+return (
+    <section className='header'>
+        <h1>To-do-list</h1>
+    </section>  
+)
```
Comment (JennieDalgren) [inline code/src/components/Header.js:14]: Here you can use an implicit return like this:
```
export const Header = () => (
    <section className='header'>
        <h1>To-do-list</h1>
    </section>  
)
```

Read more about[ explicit vs implicit returns here](https://dev.to/samanthaming/es6-arrow-functions-cheatsheet-1cn)

File: code/src/components/ItemList.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+         
+    return (
+   <section className="task-container"> 
+   <AddTaskForm />
+   {list.items.map((item, index) => (
+       <Task itemIndex={index}></Task>   
+   ))}
+   <ListSummary />
+   </section>
+    )
```
Comment (JennieDalgren) [inline code/src/components/ItemList.js]: Fix indentation!

File: code/src/components/ItemList.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+export const ItemList = () => {
+
+    const list = useSelector(store => store.todolist.list)
+         
+    return (
+   <section className="task-container"> 
+   <AddTaskForm />
+   {list.items.map((item, index) => (
+       <Task itemIndex={index}></Task>
```
Comment (JennieDalgren) [inline code/src/components/ItemList.js]: Don't forget the `key` prop

File: code/src/components/ListSummary.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+export const ListSummary = () => {
+
+    const list = useSelector(store => store.todolist.list)
+    const numDone =list.items.filter(item => item.done).length
+
+    return <section>
+        <h4>
+            {numDone} of {list.items.length} tasks complete</h4>
+    </section>
```
Comment (JennieDalgren) [inline code/src/components/ListSummary.js]: indentation!

File: code/src/components/Task.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+export const Task = ({ itemIndex }) => {
+
+    const item = useSelector((store) => store.todolist.list.items[itemIndex]);
+    const dispatch = useDispatch()
+    
+    const onRemoveClicked = event => {
+        dispatch(
+            todolist.actions.remove({
+                itemIndex: itemIndex
```
Comment (JennieDalgren) [inline code/src/components/Task.js:15]: when key and value is the same `itemIndex: itemIndex` your can write it just once `itemIndex`

File: code/src/components/Task.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+            })
+        )
+    }
+
+    const handleOnChange = event => {
+        dispatch(
+            todolist.actions.setDone({
+                itemIndex: itemIndex,
```
Comment (JennieDalgren) [inline code/src/components/Task.js:23]: same here as above with key and value

File: code/src/components/Task.js [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+  
+ return <section>
+    <div className="card">
+        <div className="checkbox-container">
+            <label className="checkbox-label">
+                <input type="checkbox"
+                    onChange={handleOnChange}
+                    checked={item.done ? "checked" : "" }></input>
+            <span class="checkbox-custom rectangular"></span>
+            </label>
```
Comment (JennieDalgren) [inline code/src/components/Task.js]: indentation!

File: code/src/pages/Home.js [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+
+const reducer = combineReducers({ todolist: todolist.reducer})
+const store = configureStore({ reducer })
+
+export const Home = () => {
+  return (
+  <Provider store={store}>
+    <main className='list-container'>
+        <Header />
+        <ItemList></ItemList>
```
Comment (JennieDalgren) [inline code/src/pages/Home.js]: this can be written as a self closing tag
`<ItemList />`

File: code/src/reducers/todolist.js [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+export const todolist = createSlice ({
+  name: "todolist",
+  initialState: initialState,
+  reducers: {
+    addTodo: (state, action) => {
+      const itemInfo = action.payload
+      state.list.items.push(itemInfo)
+    },
+    setDone: (state, action) => {
+      const { itemIndex, done } =action.payload
```
Comment (JennieDalgren) [inline code/src/reducers/todolist.js]: space after `=`

File: code/src/reducers/todolist.js [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+      state.list.items[itemIndex].done = done
+
+    },
+    remove: (state, action) => {
+      const { itemIndex } =action.payload
```
Comment (JennieDalgren) [inline code/src/reducers/todolist.js]: space after `=`

**Review by JennieDalgren — CHANGES_REQUESTED** [review]
Good job with this project!

Regarding your question about seeing the "flow" of data. Try to play around and look in the redux dev tool to see whats going on with the actions and the state. 

Make sure to fix up your indentation, remember that we use 2 space indentation at Technigo. But most importantly - be consistent! Reqruiters or others looking at your code will think it's messy and that will sadly have an impact of the overall evaluation of your code.
When indentation is fixed, I can approve this! 👍 

Keep up the good work!

**Comment by Leidstedt** [comment]
Hi! Now the page is updated. Do I send a new pull request?

**Review by JennieDalgren — APPROVED** [review]
Perfect! All good 🎉

**Pull request closed without merging**